tools/clang/test/HLSLFileCheck/hlsl/types/struct/anonymous-strictudt.hlsl

```diff
@@ -0,0 +1,23 @@
+// RUN: %dxc -E main -T vs_6_0 -strict-udt-casting %s | FileCheck %s
+
+// Tests declarations and uses of anonymous structs.
+
+// CHECK: call i32 @dx.op.loadInput.i32
+// CHECK: call %dx.types.CBufRet.i32 @dx.op.cbufferLoadLegacy.i32
+// CHECK: add nsw i32
+// CHECK: call void @dx.op.storeOutput.i32
+
+typedef struct { int x; } typedefed;
+
+struct { int x; } global;
+struct Outer
+{
+    struct { int x; } field;
+};
+
+int main(Outer input : IN) : OUT
+{
+    struct { int x; } local = input.field;
+    typedefed retval = (typedefed)local;
+    return retval.x + global.x;
```

Comment:
I think that implicitly casting from a named struct to an unnamed should succeed. Is that correct? Is there a test for that?

Reply:
Good catch, I don't think there is one.  I thought I hit this case, but perhaps it was some manual testing I was doing when making the change.  I'll see if I can add something here.

Reply:
Actually, there are cases in the `conversions-between-type-shapes-strictudt.hlsl` VerifierTest:

```
    // implicit casting to anonymous struct of identical layout ok
    struct { int a, b; } anon2_1 = s2;
```
and
```
    // implicit casting to anonymous struct of subset layout is not legal
    struct { int a, b; } anon2_2;
    anon2_2 = derivedfrom_s2;                               /* expected-error {{cannot implicitly convert from 'DerivedFrom_S2' to 'struct (anonymous struct at }} */
    struct { int a, b; } anon2_3;
    anon2_3 = s4;                                           /* expected-error {{cannot implicitly convert from 'S4' to 'struct (anonymous struct at }} */
```